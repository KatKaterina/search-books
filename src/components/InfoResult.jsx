import React from 'react';
import { Row } from 'react-bootstrap';

const InfoResult = (props) => {
  const { totalItems, loading } = props;
  const info = (load) => {
    if (load === '' || load === 'loading') {
      return null;
    }
    return load === 'success' ? `Found ${totalItems} results` : 'network error';
  };

  return (
    <Row className="justify-content-center mb-4 mt-4">
      {info(loading)}
    </Row>
  );
};

export default InfoResult;
