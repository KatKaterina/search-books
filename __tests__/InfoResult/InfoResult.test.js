import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect.js';
import React from 'react';
import InfoResult from '../../src/components/InfoResult';

describe('FormSearch', () => {
  it('render InfoResult ', () => {
    const component = render(
      <InfoResult  totalItems={0} loading={''} />
    );
    expect(component).toMatchSnapshot();

    expect(screen.queryByDisplayValue('Found 0 results')).toBeNull();
    expect(screen.queryByDisplayValue('network error')).toBeNull();
  });
  it('render InfoResult with success load', () => {
    const component = render(
      <InfoResult  totalItems={2} loading={'success'} />
    );
    expect(component).toMatchSnapshot();
    expect(screen.getByText('Found 2 results')).toBeInTheDocument();
    expect(screen.queryByDisplayValue('network error')).toBeNull();
  });
  it('render InfoResult with failed load', () => {
    const component = render(
      <InfoResult  totalItems={3} loading={'failed'} />
    );
    expect(component).toMatchSnapshot();
    expect(screen.queryByDisplayValue('Found 3 results')).toBeNull();
    expect(screen.getByText('network error')).toBeInTheDocument();
  });
});
