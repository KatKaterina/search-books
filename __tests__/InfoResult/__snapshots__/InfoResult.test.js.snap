// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSearch render InfoResult  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="justify-content-center mb-4 mt-4 row"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="justify-content-center mb-4 mt-4 row"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FormSearch render InfoResult with failed load 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="justify-content-center mb-4 mt-4 row"
      >
        network error
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="justify-content-center mb-4 mt-4 row"
    >
      network error
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FormSearch render InfoResult with success load 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="justify-content-center mb-4 mt-4 row"
      >
        Found 2 results
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="justify-content-center mb-4 mt-4 row"
    >
      Found 2 results
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
